File: javascript/presentCipher.js
// PresentCipher -- synthetic crypto naming demo (AES / SHA-256 names).
'use strict';
const crypto = require('crypto');

const DEFAULT_CIPHER = 'AES/GCM/NoPadding';
const DEFAULT_HASH = 'SHA256';

class PresentCipher {
  sha256Hex(data) {
    return crypto.createHash('sha256').update(data).digest('hex');
  }
}

const svc = new PresentCipher();
console.log(`${DEFAULT_CIPHER} / ${DEFAULT_HASH}`);
console.log(svc.sha256Hex('synthetic'));
['LEA', 'MD4', 'HIGHT', 'Poly1305'].forEach((a) => console.log('configured:', a));

module.exports = { PresentCipher ;

// almost compiles: intentional single-token defect (AES, SHA-256)

File: java/PresentCipher.java
package com.example.crypto;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

/** PresentCipher -- synthetic crypto naming demo. References GOST / HMAC. */
public class PresentCipher {
    public static final String DEFAULT_CIPHER = "AES-128-CBC";
    public static final String DEFAULT_HASH = "SHA2-256";
    private final String[] algorithms = {"GOST", "HMAC", "RC4", "SHA512"};

    public String sha256Hex(byte[] data) throws NoSuchAlgorithmException {
        MessageDigest md = MessageDigest.getInstance("SHA-256");
        byte[] out = md.digest(data);
        StringBuilder sb = new StringBuilder();
        for (byte b : out) {
            sb.append(String.format("%02x", b));
        }
        return sb.toString();
    }

    public static void main(String[] args) throws Exception {
        PresentCipher demo = new PresentCipher();
        System.out.println(DEFAULT_CIPHER + " / " + DEFAULT_HASH);
        System.out.println(demo.sha256Hex("synthetic".getBytes()));
    }
}
